Aplib.Core/ExposedQueue.cs

```diff
@@ -169,17 +171,15 @@ public bool Contains(T item)
             return false;
         }
 
-        /// <inheritdoc/>
-        public bool Remove(T item)
-        {
-            for (int i = 0; i < Count; i++)
-                if (this[i]!.Equals(item))
-                {
-                    this[i] = default!;
-                    return true;
-                }
-            return false;
-        }
+        /// <summary>
+        /// DOES NOT DO ANYTHING.
+        /// </summary>
+        /// <returns>False.</returns>
+        /// <remarks>
+        /// Method is there to comply with the ICollection interface. 
+        /// Does not actually do anything since this is a queue, and queues do not allow removal of specific elements.
+        /// </remarks>
+        public bool Remove(T item) => false;
```

Comment:
Ah, Remove is necessary for the ICollection. Didn't realize that. Maybe we should implement it then, but without leaving a gap. It's a bit of a hassle but I don't think it should do nothing actually. @joachimdekker what do you think?

Reply:
Yes. Leaving this with a false is bad practice, and will confuse people. You could better throw a NotImplementedException and discard the issue Sonar raises.

As for a better implementation, it should be possible to remove something from this queue if we make it an ICollection. However, don't think too hard about the complexity, we could just call `_array.Remove(item)` and update the `_head` and `Count` accordingly. This makes this method O(n), or worse, but that is allowed, since we have a Queue datastructure.